warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java

```diff
@@ -1344,7 +1395,163 @@ public void close() {
+                            // Validate the query against the rule's criteria.
+                            result.addRuleResult(rule.validate(validationConfig));
+                            // Remove the rule from the underlying list so that it is not executed again later.
+                            ruleIter.remove();
+                        }
+                    } catch (Exception e) {
+                        QueryException exception = new QueryException("Error occurred when validating against rule " + rule.getName(), e);
+                        result.setException(exception);
+                        return result;
+                    }
+                }
+            }
+        }
+
+        // Some rules expect to validate a JEXL query. Even if the query was originally provided in a syntax other than JEXL, e.g. LUCENE, we should validate
+        // the JEXL version of the query against the remaining rules.
+
+        // Parse the query to JEXL.
+        String jexlQuery;
+        try {
+            jexlQuery = getJexlQueryString(settings);
+            config.setQueryTree(JexlASTHelper.parseAndFlattenJexlQuery(jexlQuery));
+        } catch (Exception e) {
+            if (log.isTraceEnabled()) {
+                log.trace("Failed to parse query to JEXL: " + System.identityHashCode(this) + '('
+                                + (this.getSettings() == null ? "empty" : this.getSettings().getId()) + ')', e);
+            }
+            QueryException exception = new QueryException("Failed to parse query as JEXL", e, DatawaveErrorCode.INVALID_SYNTAX_PARSE_ERROR.getErrorCode());
+            result.setException(exception);
+            return result;
+        }
+
+        // Normalize the JEXL query on a very basic level, and apply the query model to the query.
+        // Extract any query options and add them to the configuration.
+        Map<String,String> optionsMap = new HashMap<>();
+        if (jexlQuery.contains(QueryFunctions.QUERY_FUNCTION_NAMESPACE + ':')) {
+            // only do the extra tree visit if the function is present
+            config.setQueryTree(QueryOptionsFromQueryVisitor.collect(config.getQueryTree(), optionsMap));
+            if (!optionsMap.isEmpty()) {
+                QueryOptionsSwitch.apply(optionsMap, config);
+            }
+        }
+
+        // Ensure any nodes with the literal on the left and the identifier on the right are re-ordered.
+        config.setQueryTree(InvertNodeVisitor.invertSwappedNodes(config.getQueryTree()));
+        // Uppercase all identifiers.
+        config.setQueryTree(ShardQueryUtils.upperCaseIdentifiers(metadataHelper, config, config.getQueryTree()));
+        // Flatten the tree.
+        config.setQueryTree(TreeFlatteningRebuilder.flatten(config.getQueryTree()));
+        // Apply the query model.
+        config.setQueryTree(ShardQueryUtils.applyQueryModel(config.getQueryTree(), config, metadataHelper.getAllFields(config.getDatatypeFilter()),
```

Comment:
Do we need to apply the query model for query validation?  There may be a good reason, I am just unaware of which rule(s) require this.

Reply:
Nevermind.... I guess this would be needed for the field existence rule to work correctly.